lib/github_changelog_generator/octo_fetcher.rb

```diff
@@ -280,19 +281,47 @@ def iterate_pages(client, method, *args)
     #
     # @return [Object] returns exactly the same, what you put in the block, but wrap it with begin-rescue block
     def check_github_response
+      attempt = 1
       begin
         value = yield
       rescue Octokit::Unauthorized => e
         Helper.log.error e.message
-        abort "Error: wrong GitHub token"
+        sys_abort("Error: wrong GitHub token")
       rescue Octokit::Forbidden => e
+        attempt += 1
+        sleep_time = exp_backoff(attempt)
+        Helper.log.warn("sleeping #{sleep_time} second")
+        sys_sleep(sleep_time)
+
         Helper.log.warn e.message
         Helper.log.warn GH_RATE_LIMIT_EXCEEDED_MSG
         Helper.log.warn @client.rate_limit
+
+        if attempt >= MAX_FORBIDDEN_RETRIES
+          sys_abort("Exceeded retry limit")
+        else
+          retry
+        end
       end
       value
     end
 
+    def sys_sleep(seconds)
+      sleep(seconds)
+    end
+
+    def sys_abort(msg)
+      abort(msg)
+    end
+
+    # Returns the exponential backoff (seconds) for this attempt number
+    #
+    # @param [Integer] attempt the attempt number
+    # @return [Integer] Exponential backoff seconds
+    def exp_backoff(attempt)
+      (2**attempt - 1) / 2
```

Comment:
Why did you decide to take an exponential algorithm and `MAX_FORBIDDEN_RETRIES = 100`? Is there some rule or it's just approximation?
 I just read about 401 error on https://developer.github.com/v3/#failed-login-limit and checked https://developer.github.com/guides/best-practices-for-integrators/
But didn't found any suggestions about the delay before next attempt.